kotlinx-coroutines-core/common/src/internal/DispatchedContinuation.kt

```diff
@@ -243,6 +220,12 @@ internal class DispatchedContinuation<in T>(
         "DispatchedContinuation[$dispatcher, ${continuation.toDebugString()}]"
 }
 
+@Suppress("NOTHING_TO_INLINE") // we need it inline to save us an entry on the stack
+internal inline fun <T> Continuation<T>.resumeCancellableWith(result: Result<T>) = when (this) {
+    is DispatchedContinuation -> resumeCancellableWith(result)
+    else -> resumeWith(result)
```

Comment:
The way the code is structured now, callers of `resumeCancellableWith` recover stack strack when needed.

Reply:
Yes. We don't recover neither from `suspend fun main` nor with `Unconfined` dispatcher, since recovery happen only when this continuation is dispatched. We need more tests on stack trace recovery.